jbpm-task-assigning/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolutionProcessor.java

```diff
@@ -0,0 +1,249 @@
+            this.executeResult = executeResult;
+        }
+
+        public boolean hasException() {
+            return exception != null;
+        }
+
+        public Exception getException() {
+            return exception;
+        }
+
+        public ExecutePlanningResult getExecuteResult() {
+            return executeResult;
+        }
+    }
+
+    /**
+     * @param delegate a TaskAssigningRuntimeDelegate instance for executing methods into the jBPM runtime.
+     * @param resultConsumer a consumer for processing the results.
+     * @param targetUserId a user identifier for using as the "on behalf of" user when interacting with the jBPM runtime.
+     * @param publishWindowSize Integer value > 0 that indicates the number of tasks to be published.
+     */
+    public SolutionProcessor(final TaskAssigningRuntimeDelegate delegate,
+                             final Consumer<Result> resultConsumer,
+                             final String targetUserId,
+                             final int publishWindowSize) {
+        checkNotNull("delegate", delegate);
+        checkNotNull("resultConsumer", resultConsumer);
+        checkNotNull("targetUserId", targetUserId);
+        checkCondition("publishWindowSize", publishWindowSize > 0);
+        this.delegate = delegate;
+        this.resultConsumer = resultConsumer;
+        this.targetUserId = targetUserId;
+        this.publishWindowSize = publishWindowSize;
+    }
+
+    /**
+     * @return true if a solution is being processed at this time, false in any other case.
+     */
+    public boolean isProcessing() {
+        return processing.get();
+    }
+
+    /**
+     * This method is invoked form a different thread for doing the processing of a solution. This method is not
+     * thread-safe and it's expected that any synchronization required between the isProcessing() and process()
+     * methods is performed by the caller. Since only one solution can be processed at time, the caller should typically
+     * execute in the following sequence.
+     * if (!solutionProcessor.isProcessing()) {
+     * solutionProcessor.process(solution);
+     * } else {
+     * //invoke at a later time.
+     * }
+     * A null value will throw an exception.
+     * @param solution a solution to process.
+     */
+    public void process(final TaskAssigningSolution solution) {
+        checkNotNull("solution", solution);
+        processing.set(true);
+        this.solution = solution;
+        solutionResource.release();
+    }
+
+    @Override
+    public void destroy() {
+        super.destroy();
+        solutionResource.release(); //un-lock in case it was waiting for a solution to process.
+    }
+
+    @Override
+    public void run() {
+        while (isAlive()) {
+            try {
+                solutionResource.acquire();
+                if (isAlive()) {
+                    doProcess(solution);
+                }
+            } catch (InterruptedException e) {
```

Comment:
If there is no need to re-throw the InterruptedException, let's set the interrupted flag via `Thread.interrupt()`.

See e.g. https://dzone.com/articles/understanding-thread-interruption-in-java

Reply:
I'll restore the interrupted status as suggested by zonar, but I believe the flag by zonar is a "false positive". 

This is what the code is doing.

        while (isAlive()) {
               //do something
            } catch (InterruptedException e) {
             //so here the code sets it self in super.destory() that will cause the 
             //the thread to have a graceful and controlled die. The interruption is  thus not hiden,
             // and imediatelly when the whild(isAlive()) condition is checked the result is 
             // false. No risks to enter in a neverending loop ingnoring the interrupted status.

                super.destroy();
                
                LOGGER.error("Solution Processor was interrupted", e);
            }
        }